pkg/utils/maputils.go

```diff
@@ -87,24 +87,22 @@ func SetDifference[K comparable, T1, T2 any](newSet map[K]T2, oldSet map[K]T1) (
 	return added, removed
 }
 
-// ConvertToSetFromMap converts a map to a set. The set is represented as a map with the key as the element of the set
-func ConvertToSetFromMap[K comparable, V any](set map[K]struct{}, source map[K]V) {
+// AddMapKeysToSet converts a map to a set. The set is represented as a map with the key as the element of the set
+func AddMapKeysToSet[K comparable, V any](set map[K]struct{}, source map[K]V) {
```

Comment:
nit: with this name change, I don't think you even need the comment any more

Same comment for `AddSliceToSet()`